state/state.go

```diff
@@ -34,6 +40,110 @@ type DownloadedObject struct {
 	Object, Filename string
 }
 
+type DynamoDBStater struct {
+	*sync.Mutex
+	Session *session.Session
+	Service string
+}
+
+func NewDynamoDBStater(session *session.Session, service string) *DynamoDBStater {
+	return &DynamoDBStater{
+		Mutex:   &sync.Mutex{},
+		Session: session,
+		Service: service,
+	}
+}
+
+// list of processed objects
+func (d *DynamoDBStater) processedObjects() (map[string]time.Time, error) {
+	objs := make(map[string]time.Time)
+
+	type Record struct {
+		S3object string    `json:"S3object"`
+		Time     time.Time `json:"Time"`
+	}
+
+	var records []Record
+
+	svc := dynamodb.New(d.Session)
+	err := svc.ScanPages(&dynamodb.ScanInput{
+		TableName: aws.String("HoneyELBAccessLogBuckets"),
+	}, func(logs *dynamodb.ScanOutput, last bool) bool {
+		recs := []Record{}
+
+		err := dynamodbattribute.UnmarshalListOfMaps(logs.Items, &recs)
+
+		// break out of function
+		if err != nil {
+			panic(fmt.Sprintf("failed to unmarshal Dynamodb Scan Items, %v", err))
```

Comment:
panic seems a little severe - are there ever transient failures that might lead to being unable to unmarshal this content? 
Maybe we can leave this as a `panic` but give some more instruction to the end user on what to do about it ("please copy/paste this into an email to support@ because this shouldn't happen!" or some such)

Reply:
Hm, yes also we should make sure that an empty table `Unmarshal`s without error, for instance I had to add this check to `FileStater` https://github.com/honeycombio/honeyaws/blob/master/state/state.go#L61-L64